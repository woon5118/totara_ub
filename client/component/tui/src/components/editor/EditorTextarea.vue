<!--
  This file is part of Totara Enterprise Extensions.

  Copyright (C) 2020 onwards Totara Learning Solutions LTD

  Totara Enterprise Extensions is provided only to Totara
  Learning Solutions LTD's customers and partners, pursuant to
  the terms and conditions of a separate agreement with Totara
  Learning Solutions LTD or its affiliate.

  If you do not have an agreement with Totara Learning Solutions
  LTD, you may not access, use, modify, or distribute this software.
  Please contact [licensing@totaralearning.com] for more information.

  @author Simon Chester <simon.chester@totaralearning.com>
  @module tui
-->

<template>
  <div class="tui-editorTextarea">
    <Textarea
      :value="value && value.content"
      class="tui-editorTextarea__textarea"
      char-length="full"
      rows="8"
      @input="handleInput"
    />
  </div>
</template>

<script>
import Textarea from 'tui/components/form/Textarea';

export default {
  components: {
    Textarea,
  },

  props: {
    value: {
      type: Object,
    },
  },

  mounted() {
    this.$emit('ready');
  },

  methods: {
    handleInput(val) {
      this.$emit('input', { content: val });
    },
  },
};
</script>

<style lang="scss">
.tui-editorTextarea {
  // stretch child input to be full height if the editor is given a specific height
  display: flex;
  // expand to full width if in horizontal flex
  width: 100%;
}
</style>
