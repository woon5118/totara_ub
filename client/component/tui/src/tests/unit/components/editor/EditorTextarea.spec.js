/**
 * This file is part of Totara Enterprise Extensions.
 *
 * Copyright (C) 2021 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise Extensions is provided only to Totara
 * Learning Solutions LTD's customers and partners, pursuant to
 * the terms and conditions of a separate agreement with Totara
 * Learning Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [licensing@totaralearning.com] for more information.
 *
 * @author Brian Barnes <brian.barnes@totaralearning.com>
 * @module tui
 */

import { shallowMount } from '@vue/test-utils';
import EditorTextarea from 'tui/components/editor/EditorTextarea';

describe('EditorTextarea', () => {
  let wrapper = {};

  beforeEach(() => {
    wrapper = shallowMount(EditorTextarea, {
      propsData: {
        value: { content: 'text value' },
      },
    });
  });

  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
