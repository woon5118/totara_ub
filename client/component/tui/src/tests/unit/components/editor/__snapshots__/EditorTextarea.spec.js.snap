// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorTextarea matches snapshot 1`] = `
<div
  class="tui-editorTextarea"
>
  <textarea-stub
    charlength="full"
    class="tui-editorTextarea__textarea"
    rows="8"
    value="text value"
  />
</div>
`;
